File: TREE/EASY/Minimum Depth of Binary Tree/Code2.py
# PROBLEM LINK:- https://leetcode.com/problems/minimum-depth-of-binary-tree/

class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        if None in [root.left, root.right]:
            return 1 + max(self.minDepth(root.left), self.minDepth(root.right))
        return 1 + min(self.minDepth(root.left), self.minDepth(root.right))

File: ARRAYS/Easy/Matrix Diagonal Sum/Code2.cpp
/*
PROBLEM LINK:- https://leetcode.com/problems/matrix-diagonal-sum/
*/

class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int x = 0, y = 0, n = mat.size();
        for(int i = 0; i < n; i++)
        {
            x += mat[i][i];
            y += mat[i][n-i-1];
        }
        
        if(n % 2 == 0)
            return x + y;
        return x + y - mat[n/2][n/2];
    }
};
